### PR TITLE
Documentation for Lib.

### PR DESCRIPTION
**Pull request by asmanur**
Documentation for the Lib module.

Please have a look at the description of the function.

### CODE REVIEW

File: lib/lib.mli [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+
+(** This section deals with parsing protocols. *)
+
+val parse_string : string -> Syntax.scr_module
+(** Parse a string into a {!Syntax.scr_module}. *)
+
+val parse : string -> Core_kernel.In_channel.t -> Syntax.scr_module
+(** Parse from an input channel. The firs parameter is the filename, for use
+    in error messages. *)
+
```
Comment (fangyi-zhou) [inline lib/lib.mli]: Section 2: Validation?

File: lib/lib.mli [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+val validate_exn : Syntax.scr_module -> verbose:bool -> unit
+(** [validate_exn module ~verbose] validates the module [module] by
+    performing standard checks. If [verbose] is set to true, output messages
+    in the process. *)
+
```
Comment (fangyi-zhou) [inline lib/lib.mli]: Section 3: Other Operations?

File: lib/lib.mli [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+(** {1 Parsing} *)
+
+(** This section deals with parsing protocols. *)
+
+val parse_string : string -> Syntax.scr_module
+(** Parse a string into a {!Syntax.scr_module}. *)
+
+val parse : string -> Core_kernel.In_channel.t -> Syntax.scr_module
+(** Parse from an input channel. The firs parameter is the filename, for use
```
Comment (fangyi-zhou) [inline lib/lib.mli]:
```suggestion
(** Parse from an input channel. The first parameter is the filename, for use
```

**Comment by asmanur** [comment]
Just saw that, will apply the changes.

**Comment by asmanur** [comment]
Done, rebased master and updated the commit.

**Pull request merged**